src/api/app/controllers/webui/watched_items_controller.rb

```diff
@@ -0,0 +1,40 @@
+class Webui::WatchedItemsController < Webui::WebuiController
+  before_action :check_user_belongs_feature_flag
+  before_action :set_item
+
+  FLASH_PER_WATCHABLE_TYPE = {
+    Package => 'package',
+    Project => 'project',
+    BsRequest => 'request'
+  }.freeze
+
+  def toggle
+    watched_item = User.session!.watched_items.find_by(watchable: @item)
+
+    if watched_item
+      watched_item.destroy
+      flash[:success] = "Removed #{FLASH_PER_WATCHABLE_TYPE[@item.class]} from the watchlist"
+    else
+      User.session!.watched_items.create(watchable: @item)
+      flash[:success] = "Added #{FLASH_PER_WATCHABLE_TYPE[@item.class]} to the watchlist"
+    end
+
+    redirect_back(fallback_location: root_path)
+  end
+
+  private
+
+  def set_item
+    @item = if params[:package]
+              Package.find_by_project_and_name(params[:project], params[:package])
+            elsif params[:project]
+              Project.find_by(name: params[:project])
```

Comment:
Did you consider [links](https://github.com/openSUSE/open-build-service/wiki/Links)?

Reply:
No, we didn't. We'll open a follow-up PR to address this.

Reply:
I'm a bit worried here as this will have influence on notifications one day-

Reply:
We added a card in our backlog for this: https://trello.com/c/9wC6V7qQ/2520-handle-links-in-watchlist